`TestServerStartAndShutdown` fails

Go 1.22, MacOS, firstly discovered in #3398.
```
2024-04-04T09:47:28.5050930Z === RUN   TestServerStartAndShutdown
2024-04-04T09:47:28.5051200Z === RUN   TestServerStartAndShutdown/no_consensus
2024-04-04T09:47:28.5052500Z     logger.go:146: 2024-04-04T09:43:01.073Z	INFO	ExtensiblePoolSize is not set or wrong, using default value	{"ExtensiblePoolSize": 20}
2024-04-04T09:47:28.5053710Z     logger.go:146: 2024-04-04T09:43:01.073Z	INFO	bad MaxPeers configured, using the default value	{"configured": 0, "actual": 100}
2024-04-04T09:47:28.5055010Z     logger.go:146: 2024-04-04T09:43:01.073Z	INFO	bad AttemptConnPeers configured, using the default value	{"configured": 0, "actual": 20}
2024-04-04T09:47:28.5055960Z     logger.go:146: 2024-04-04T09:43:01.073Z	INFO	node started	{"blockHeight": 0, "headerHeight": 0}
2024-04-04T09:47:28.5056860Z     logger.go:146: 2024-04-04T09:43:01.073Z	INFO	node reached synchronized state, starting services
2024-04-04T09:47:28.5057800Z     logger.go:146: 2024-04-04T09:43:01.074Z	INFO	new peer connected	{"addr": "0.0.0.0:0", "peerCount": 1}
2024-04-04T09:47:28.5058020Z     server_test.go:98: 
2024-04-04T09:47:28.5058830Z         	Error Trace:	/Users/runner/work/neo-go/neo-go/pkg/network/server_test.go:98
2024-04-04T09:47:28.5059170Z         	Error:      	Should be true
2024-04-04T09:47:28.5059600Z         	Test:       	TestServerStartAndShutdown/no_consensus
2024-04-04T09:47:28.5060310Z     logger.go:146: 2024-04-04T09:43:01.084Z	INFO	shutting down server	{"peers": 1}
2024-04-04T09:47:28.5060510Z peer dropped: server shutdown
2024-04-04T09:47:28.5061700Z     logger.go:146: 2024-04-04T09:43:01.088Z	WARN	peer disconnected	{"addr": "0.0.0.0:0", "error": "server shutdown", "peerCount": 0}
2024-04-04T09:47:28.5062070Z === RUN   TestServerStartAndShutdown/with_consensus
2024-04-04T09:47:28.5063380Z     logger.go:146: 2024-04-04T09:43:01.088Z	INFO	ExtensiblePoolSize is not set or wrong, using default value	{"ExtensiblePoolSize": 20}
2024-04-04T09:47:28.5064680Z     logger.go:146: 2024-04-04T09:43:01.088Z	INFO	bad MaxPeers configured, using the default value	{"configured": 0, "actual": 100}
2024-04-04T09:47:28.5065990Z     logger.go:146: 2024-04-04T09:43:01.088Z	INFO	bad AttemptConnPeers configured, using the default value	{"configured": 0, "actual": 20}
2024-04-04T09:47:28.5066880Z     logger.go:146: 2024-04-04T09:43:01.088Z	INFO	node started	{"blockHeight": 0, "headerHeight": 0}
2024-04-04T09:47:28.5067760Z     logger.go:146: 2024-04-04T09:43:01.088Z	INFO	node reached synchronized state, starting services
2024-04-04T09:47:28.5068680Z     logger.go:146: 2024-04-04T09:43:01.088Z	INFO	new peer connected	{"addr": "0.0.0.0:0", "peerCount": 1}
2024-04-04T09:47:28.5069380Z     logger.go:146: 2024-04-04T09:43:01.088Z	INFO	shutting down server	{"peers": 1}
2024-04-04T09:47:28.5069580Z peer dropped: server shutdown
2024-04-04T09:47:28.5070780Z     logger.go:146: 2024-04-04T09:43:01.089Z	WARN	peer disconnected	{"addr": "0.0.0.0:0", "error": "server shutdown", "peerCount": 0}
2024-04-04T09:47:28.5071050Z === RUN   TestServerStartAndShutdown/double_start
2024-04-04T09:47:28.5072330Z     logger.go:146: 2024-04-04T09:43:01.089Z	INFO	ExtensiblePoolSize is not set or wrong, using default value	{"ExtensiblePoolSize": 20}
2024-04-04T09:47:28.5073550Z     logger.go:146: 2024-04-04T09:43:01.089Z	INFO	bad MaxPeers configured, using the default value	{"configured": 0, "actual": 100}
2024-04-04T09:47:28.5074850Z     logger.go:146: 2024-04-04T09:43:01.089Z	INFO	bad AttemptConnPeers configured, using the default value	{"configured": 0, "actual": 20}
2024-04-04T09:47:28.5075730Z     logger.go:146: 2024-04-04T09:43:01.089Z	INFO	node started	{"blockHeight": 0, "headerHeight": 0}
2024-04-04T09:47:28.5076600Z     logger.go:146: 2024-04-04T09:43:01.089Z	INFO	node reached synchronized state, starting services
2024-04-04T09:47:28.5077230Z     logger.go:146: 2024-04-04T09:43:01.089Z	INFO	node server already started
2024-04-04T09:47:28.5077930Z     logger.go:146: 2024-04-04T09:43:01.089Z	INFO	shutting down server	{"peers": 0}
2024-04-04T09:47:28.5078230Z === RUN   TestServerStartAndShutdown/double_shutdown
2024-04-04T09:47:28.5079520Z     logger.go:146: 2024-04-04T09:43:01.089Z	INFO	ExtensiblePoolSize is not set or wrong, using default value	{"ExtensiblePoolSize": 20}
2024-04-04T09:47:28.5080740Z     logger.go:146: 2024-04-04T09:43:01.090Z	INFO	bad MaxPeers configured, using the default value	{"configured": 0, "actual": 100}
2024-04-04T09:47:28.5082030Z     logger.go:146: 2024-04-04T09:43:01.090Z	INFO	bad AttemptConnPeers configured, using the default value	{"configured": 0, "actual": 20}
2024-04-04T09:47:28.5082900Z     logger.go:146: 2024-04-04T09:43:01.090Z	INFO	node started	{"blockHeight": 0, "headerHeight": 0}
2024-04-04T09:47:28.5083840Z     logger.go:146: 2024-04-04T09:43:01.090Z	INFO	node reached synchronized state, starting services
2024-04-04T09:47:28.5084560Z     logger.go:146: 2024-04-04T09:43:01.090Z	INFO	shutting down server	{"peers": 0}
2024-04-04T09:47:28.5084880Z --- FAIL: TestServerStartAndShutdown (0.02s)
2024-04-04T09:47:28.5085350Z     --- FAIL: TestServerStartAndShutdown/no_consensus (0.02s)
2024-04-04T09:47:28.5085850Z     --- PASS: TestServerStartAndShutdown/with_consensus (0.00s)
2024-04-04T09:47:28.5086400Z     --- PASS: TestServerStartAndShutdown/double_start (0.00s)
2024-04-04T09:47:28.5086910Z     --- PASS: TestServerStartAndShutdown/double_shutdown (0.00s)
```

--- a/pkg/network/server_test.go
+++ b/pkg/network/server_test.go
@@ -95,8 +95,10 @@ func TestServerStartAndShutdown(t *testing.T) {
 		s.register <- p
 		require.Eventually(t, func() bool { return 1 == s.PeerCount() }, time.Second, time.Millisecond*10)
 
-		assert.True(t, s.transports[0].(*fakeTransp).started.Load())
 		require.True(t, s.started.Load())
+		require.Eventually(t, func() bool {
+			return s.transports[0].(*fakeTransp).started.Load()
+		}, 2*time.Second, 200*time.Millisecond)
 		assert.Nil(t, s.txCallback)
 
 		s.Shutdown()
@@ -115,6 +117,7 @@ func TestServerStartAndShutdown(t *testing.T) {
 		s.Start()
 		p := newLocalPeer(t, s)
 		s.register <- p
+		require.Eventually(t, func() bool { return 1 == s.PeerCount() }, time.Second, time.Millisecond*10)
 
 		assert.True(t, s.services["fake"].(*fakeConsensus).started.Load())
 		require.True(t, s.started.Load())

